python/packages/autogen-agentchat/src/autogen_agentchat/base/_team.py

```diff
@@ -1,17 +1,20 @@
-from typing import Any, Mapping, Protocol
-
+from typing import Any, Mapping
+from abc import ABC, abstractmethod
```

Comment:
@jackgerrits , 
Wanted to start this PR to move Team and Memory to use ABC as them also need to be declarative/serializable